src/App.tsx

```diff
@@ -1,39 +1,70 @@
-import React from 'react';
-import './App.scss';
+import { Component } from 'react';
+import { Clock } from './components/Clock';
 
-function getRandomName(): string {
-  const value = Date.now().toString().slice(-4);
+import './App.scss';
 
-  return `Clock-${value}`;
+interface State {
+  clockName: string;
+  id: number;
+  hasClock: boolean;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends Component<{}, State> {
+  state = {
+    clockName: 'Clock-0',
+    id: 0,
+    hasClock: true,
+  };
+
+  componentDidMount() {
+    const timerId = window.setInterval(() => {
+      this.getRandomName();
+    }, 3300);
+
+    this.setState({ id: timerId });
+
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
+      this.setState({ hasClock: false });
+    });
+
+    document.addEventListener('click', () => {
+      this.setState({ hasClock: true });
+    });
+  }
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentWillUnmount() {
+    window.clearInterval(this.state.id);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.removeEventListener('contextmenu', () => {
+      this.setState({ hasClock: false });
+    });
```

Comment:
function won't be removed in this way
for deleting handlers you need to pass the same link to the same function as the second argument
for now you created separate function, and pass it here, it won't delete existing handler